SPEC_ASSUMPTIONS.md

```diff
@@ -0,0 +1,122 @@
+
+If I make div#bee not hidden then I get ‘Hello world words’. I think that Chrome’s result makes the most sense here, and so maybe ‘direct’ in the spec should be defined, or rule 2A should be changed?
+
+#### Related GitHub Issues:
+- https://github.com/w3c/accname/issues/48
+- https://github.com/w3c/accname/issues/57
+## A.2
+Step 2A: ‘hidden’ is defined as ‘not visible, perceivable, or interactive to any user. An element is considered hidden if it or any one of its ancestor elements is not rendered or is explicitly hidden’.
+
+In the context of HTML, CSS we consider an element to be hidden if it:
+- Has style ‘visibility: hidden’
+- Has an offsetHeight and offsetWidth of 0
+- Has ‘hidden’ or ‘aria-hidden=true’ attributes
+- Has any ancestors for which any of (1-3) apply
+- Is not focusable
+
+#### Related GitHub Issues
+- https://github.com/w3c/accname/issues/57
+## A.3
+Step 2A: We assume that `<option>`s shouldn’t be considered ‘hidden’ if they are part of the text alternative computation for a `<select>` (as handled by 2E role=combobox, listbox).
+
+The labelled chunk of code in 2A `isHidden()` specifically addresses the fact that Chrome considers `<option>` elements (even if selected) to be hidden, according to offsetWidth & Height == 0.
+
+## B.1
+Step 2B: ‘already part of an aria-labelledby traversal’ does this mean that any parents have been referenced by aria-labelledby or simply that the referenced node does not go on to aria-labelledby reference another?
+```html
+<div id='bar' hidden> and some more words</div>
+<div id='foo' hidden>
+    Some words 
+    <div aria-labelledby='bar'></div>
+</div>
+<div aria-labelledby='foo'></div>
+```
+Does div#foo’s aria-labelledby=’bar’ count as being part of an aria-labelledby traversal? Chrome says no, and for this example acc-name of div#foo = ‘Some words and some more words’.
+
+We assume that a node must be directly referenced by aria-labelledby to be part of an aria-labelledby traversal.
+#### Related GitHub Issues
+- https://github.com/w3c/accname/issues/79
+
+## C.1
+We assume that context.partOfName is true implies that the current node is being traversed due to recursion. We can make this assumption because partOfName is set to true any time the algorithm is called recursively.
+## D.1
+We assume that the [HTML Accessibility API Mappings spec](https://www.w3.org/TR/html-aam-1.0/#accessible-name-and-description-computation) lists all instances in which native elements or attributes define a text alternative. It is this document that we used as a guide to implement 2D.
+## E.1
```

Comment:
The assumptions for `2E` don't have any references. Do no issues on the spec exist for thesse? Should they?

Reply:
Added links to my own contributions in new commit